### PR TITLE
Refactor: Rename "More info dialog" to "Details view" for clearer tra…

### LINKED ISSUE #22517: Rename "More info dialog" to just "Details view" to streamline translations
### Describe the issue you are experiencing

Translating "More info dialog" into German is almost impossible without creating some word monster like

     "Weitere Informationen"-Dialogfeld

This is simply no working solution to cover even the very few string variations, so I came up with just calling this

     Detailansicht

instead which works remarkably well in all contexts. In English this translates back to

     Details View

 which also fits much better for complex lighting controls like the WLED example here:

Screenshot 2024-10-24 14 29 11

That isn't "more information", it's really a more detailed view to interact with that device. Similar with locations that open a map or sensors that show more detailed history graphs.

Also this is just a view, not a dialog with "Submit", "Delete" or"Cancel" buttons either.

There are just six strings that need to be updated for this change:

https://github.com/home-assistant/frontend/blob/fc9a0958d44475335ce30fc4c8099fab113b79e8/src/translations/en.json#L1428

https://github.com/home-assistant/frontend/blob/fc9a0958d44475335ce30fc4c8099fab113b79e8/src/translations/en.json#L5416

https://github.com/home-assistant/frontend/blob/fc9a0958d44475335ce30fc4c8099fab113b79e8/src/translations/en.json#L5770

And the following three where "More info dialog" can be shortened to "Details view":

https://github.com/home-assistant/frontend/blob/fc9a0958d44475335ce30fc4c8099fab113b79e8/src/translations/en.json#L5381

https://github.com/home-assistant/frontend/blob/fc9a0958d44475335ce30fc4c8099fab113b79e8/src/translations/en.json#L5901

https://github.com/home-assistant/frontend/blob/fc9a0958d44475335ce30fc4c8099fab113b79e8/src/translations/en.json#L6162

And finally this one, where it's even inconsistent in English with "info" instead of "More info":

https://github.com/home-assistant/frontend/blob/311f221387c30b0a2c56015aae641131c134b128/src/translations/en.json#L1231

All other occurrences of "More info" or "More information" open the Help etc. and should not be changed.

### What version of Home Assistant Core has the issue?

core-2024.10.3

### PR DESCRIPTION
**Pull request by iemafzalhassan**
For issue #22517, here’s a PR description that aligns with the requested format:

---

## Proposed change
This PR renames the "More info dialog" to "Details view" across the codebase to streamline terminology and simplify translations. This adjustment aligns with the naming conventions used in other components, providing a clearer and more cohesive user experience.

## Type of change
- [x] Code quality improvements to existing code or addition of tests

## Additional information
- This PR fixes or closes issue: fixes #22517
- This PR is related to issue or discussion: #22517
- Link to documentation pull request: N/A

## Checklist
- [x] The code change is tested and works locally.
- [x] There is no commented-out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user-exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

---

#### Contributor GitHub: [@iemafzalhassan](https://github.com/iemafzalhassan)

### CODE REVIEW

**Review by iemafzalhassan — COMMENTED** [review]
**Review Feedback:**

Great job with this PR! 👍 The update to rename "More info dialog" to "Details view" makes the UI more consistent and user-friendly. Here’s some quick feedback:

1. **Description Clarity**: The PR description is well-organized and explains the reasoning behind the terminology change.
2. **Documentation**: It’s a good idea to ensure any related documentation is updated for clarity, so users have no issues navigating.
3. **Testing**: Please verify that this change has been tested across components where the term is used to avoid any unexpected UI inconsistencies.

Once these are addressed, I’d be happy to approve the PR. Nice work!

**Comment by piitaya** [comment]
Hi 👋 Thank you for the contribution. The "more info" terms is well know for HA users. It's not something we can change easily. I agree "more info dialog" is not a good term. I will discuss that point with the UX/UI team.

Also, "detail view" is not the right term because we already have "view" in the dashboard and it's another element.

**Comment by iemafzalhassan** [comment]
Hi @piitaya ,

There is a concern about using "detail view" due to the existing. Perhaps we can brainstorm alternatives that avoid confusion, a more accurate description than the "view" terminal while still pro "more info dialog." 

I’d be happy to collaborate on finding a term that is both user-familiar and clear. I'm looking forward to hearing your thoughts!

File: src/translations/en.json [hunk 1]
```diff
@@ -6159,7 +6159,7 @@
             },
             "tile": {
               "name": "Tile",
-              "description": "The tile card gives you a quick overview of your entity. The card allow you to toggle the entity, show the more info dialog or custom actions.",
+              "description": "he tile card gives you a quick overview of your entity. The card allows you to toggle the entity, show the details view, or perform custom actions.",
```
Comment (bramkragten) [inline src/translations/en.json:6162]:
```suggestion
              "description": "The tile card gives you a quick overview of your entity. The card allows you to toggle the entity, show the details view, or perform custom actions.",
```

**Comment by piitaya** [comment]
After discussing with the UX/UI team, we find that there is no reason to change for now. The "more info" wording is only used in few places (mostly for dashboard actions). We will need better research / UX writing audit before we make sweeping changes to our vocabulary such as this.

**Comment by iemafzalhassan** [comment]
This was my first real contribution, which I was expecting to get merged. Bytheway thanked! @piitaya for your time and support.

**Pull request closed without merging**